src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,6 +45,14 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        String[] splitAddress = trimmedAddress.split(ADDRESS_SPLIT_DELIMITER);
+        if (splitAddress.length < 4) { //Sanity check
```

Comment:
Sanity check is good practice! Would you want to consider length of more than 4 legal though?